lib/danger/request_source/github.rb

```diff
@@ -153,9 +176,148 @@ def delete_old_comments!(except: nil, danger_id: 'danger')
         end
       end
 
+      def submit_inline_comments!(warnings: [], errors: [], messages: [], markdowns: [], previous_violations: [], danger_id: 'danger')
+        filter = proc do |v|
+          v.file && v.line
+        end
+
+        # Avoid doing any fetchs if there's no inline comments
+        return if (warnings + errors + messages).select(&filter).empty?
```

Comment:
I've thinking about this line, and with the non-sticky comments it's no longer effective. This might end up leaving resolved violations without updating.